src/main/java/eu/europeana/metis/sandbox/service/workflow/IndexingServiceImpl.java

```diff
@@ -4,36 +4,50 @@
 
 import eu.europeana.indexing.Indexer;
 import eu.europeana.indexing.IndexingProperties;
+import eu.europeana.indexing.exception.IndexerRelatedIndexingException;
 import eu.europeana.indexing.exception.IndexingException;
+import eu.europeana.indexing.tiers.model.TierResults;
 import eu.europeana.metis.sandbox.common.exception.DatasetIndexRemoveException;
 import eu.europeana.metis.sandbox.common.exception.RecordProcessingException;
 import eu.europeana.metis.sandbox.domain.Record;
 import eu.europeana.metis.sandbox.domain.RecordInfo;
 import java.io.IOException;
 import java.util.Date;
 import javax.annotation.PreDestroy;
+
+import eu.europeana.metis.sandbox.service.record.RecordService;
 import org.springframework.stereotype.Service;
 
 @Service
 class IndexingServiceImpl implements IndexingService {
 
   private final Indexer publishIndexer;
+  private final RecordService recordService;
 
-  public IndexingServiceImpl(Indexer publishIndexer) {
+  public IndexingServiceImpl(Indexer publishIndexer, RecordService recordService) {
     this.publishIndexer = publishIndexer;
+    this.recordService = recordService;
   }
 
   @Override
   public RecordInfo index(Record recordToIndex) {
     requireNonNull(recordToIndex, "Record must not be null");
+    TierResults tierCalculations;
 
     try {
-      publishIndexer.index(recordToIndex.getContentInputStream(),
+      tierCalculations = publishIndexer.indexAndGetTierCalculations(recordToIndex.getContentInputStream(),
               new IndexingProperties(new Date(), false, null, false, true));
```

Comment:
This is out of scope of this ticket I just noticed ```IndexingProperties```. Please add a TODO: here please, to many different boolean parameters indicates many posibilities to create this object. This has to be refactored in a way that code is readable.

Reply:
Added